lms/djangoapps/gating/signals.py

```diff
@@ -21,4 +22,18 @@ def evaluate_subsection_gated_milestones(**kwargs):
     """
     subsection_grade = kwargs['subsection_grade']
     gating_api.evaluate_prerequisite(kwargs['course'], subsection_grade, kwargs.get('user'))
-    gating_api.evaluate_entrance_exam(kwargs['course'], subsection_grade, kwargs.get('user'))
+
+
+@receiver(COURSE_GRADE_CHANGED)
+def evaluate_course_gated_milestones(**kwargs):
```

Comment:
Entrance exams are at the chapter level, not at the subsection level.  Hence, it requires listening to the `COURSE_GRADE_CHANGED` signal.